src/distPlus/distribution/AlgebraicShapeCombination.re

```diff
@@ -219,35 +226,21 @@ let toDiscretePointMassesFromDiscrete = (s: DistTypes.xyShape): pointMassesWithM
   let variances: array(float) = Belt.Array.makeUninitializedUnsafe(n);
 
   for (i in 0 to n - 1) {
-    let _ =
-      Belt.Array.set(
-        masses,
-        i,
-        ys[i]
-      );
-
-    let _ =
-      Belt.Array.set(
-        means,
-        i,
-        xs[i]
-      );
-
-    let _ =
-      Belt.Array.set(
-        variances,
-        i,
-        0.0
-      );
-    ();
+    Belt.Array.set(masses, i, ys[i]) |> ignore;
+    Belt.Array.set(means, i, xs[i]) |> ignore;
+    Belt.Array.set(variances, i, 0.0) |> ignore;
```

Comment:
Why didn't you use Belt.Array.map or Belt.Array. makeBy here? You could have still cut off the ends easily enough. 

https://bucklescript.github.io/bucklescript/api/Belt.Array.html